### PR TITLE
Build: Disable AppendOnly if available

### PR DESCRIPTION
**Pull request by iyerr3**

### CODE REVIEW

**Comment by iyerr3** [comment]
Accidently created this PR. This should be ready to review in a couple of days.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/661/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/663/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/665/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/666/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/674/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/675/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/676/

File: src/ports/postgres/modules/utilities/control.py_in [hunk 1]
```diff
@@ -158,6 +159,61 @@ class MinWarning(ContextDecorator):
...
+        return ','.join(['{0}={1}'.format(k, v)
+                        for k, v in self.storage_options_dict.iteritems()])
+
+    def __enter__(self):
+        try:
+            _storage_options_str = plpy.execute(
+                "show gp_default_storage_options")[0]["gp_default_storage_options"]
+            self._parse_gp_default_storage_options(_storage_options_str)
+
+            # Set APPENDONLY=False after backing up existing value
```
[the remark this answers is not in the document]
Comment (iyerr3) [reply src/ports/postgres/modules/utilities/control.py_in]: Yes, the comment is incorrect. I will change it to "Set APPENDONLY to input value"

File: src/ports/postgres/modules/utilities/control.py_in [hunk 2]
```diff
@@ -158,6 +159,61 @@ class MinWarning(ContextDecorator):
...
+        except plpy.SPIError:
+            self.guc_exists = False
+        finally:
+            return self
+
+    def __exit__(self, *args):
+        if args and args[0]:
+            # an exception was raised in code. We return False so that any
+            # exception is re-raised after exit. The transaction will not
+            # commit leading to reset of parameter value.
```
[the remark this answers is not in the document]
Comment (iyerr3) [reply src/ports/postgres/modules/utilities/control.py_in]: The GUC value should also reset if transaction does not commit. Ideally, this context manager should be called at the highest level. However, as you mention, it's possible it's not used correctly, with the raised exception caught and ignored. I have updated the code to perform the reset before re-raising the exception.

File: src/ports/postgres/modules/utilities/control.py_in (hunk 1 above)
[the remark this answers is not in the document]
Comment (reductionista) [reply src/ports/postgres/modules/utilities/control.py_in]: Rest looks good, I will approve as soon as you update this comment.

**Pull request closed without merging**